### ISSUE TITLE
TOML: QuoteMapKeys

### ISSUE THREAD

**Issue by 89z** [TomWright/dasel#56]
If you have this JSON:

~~~json
{
   "This is a": "TOML document",
   "name": "Tom Preston-Werner",
   "title": "TOML Example"
}
~~~

you get this TOML:

~~~toml
"This is a" = "TOML document"
name = "Tom Preston-Werner"
title = "TOML Example"
~~~

Current implementation is here:

https://github.com/TomWright/dasel/blob/ba2b3c0dac195b1f0b7cfaf5250c9dd55a33b0a3/internal/storage/toml.go#L29-L33

the TOML library has an option to always quote the keys:

https://pkg.go.dev/github.com/pelletier/go-toml#Encoder.QuoteMapKeys

that would produce this output:

~~~
"This is a" = "TOML document"
"name" = "Tom Preston-Werner"
"title" = "TOML Example"
~~~

basically the code could be changed to this:

~~~
enc := toml.NewEncoder(buf).QuoteMapKeys(quote)
~~~

where `quote` is a boolean that can be set by the user.

**Comment by TomWright** [comment]
Hey @89z,

Thanks for your contribution!

I'm find with this change happening, but I'm wondering what the benefit of this is. Does this solve some kind of issue or is it purely cosmetic?

**Comment by 89z** [comment]
Its just cosmetic - it just seems awkward to me. If you have some keys that need quote, seem better to just quote all of them. But personal taste I guess

**Comment by TomWright** [comment]
Just making sure I understood that's all 🙂 
I'll get around to making this change soon!

**Comment by TomWright** [comment]
Hey @89z,

I've got a PR for this change here: https://github.com/TomWright/dasel/pull/58

I'm not sure I want this to be the default functionality (since it isn't the default for many toml implementations) but I'm happy to enable it with a feature flag.

I'm closing this issue but please head over to the PR and submit any feedback you may have.
